cpp/Agents/EvolutionStrategy.hpp

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
+#include "../environment.hpp"
+
+using namespace mlpack;
+using namespace std;
+using namespace arma;
+using namespace gym;
+
+class EvolutionStrategyAgent{
+ public:
+  mat model;
+  size_t frameW, frameH, num_actions;
+
+  EvolutionStrategyAgent(size_t frameW,
+                    size_t frameH,
+                    size_t num_actions)
+  :
+  frameW(frameW),
+  frameH(frameH),
+  num_actions(num_actions)
+  {
+    model = randn(num_actions, frameW * frameH * 3);
+  }
+
+  void Play(string environment,
+            string host,
+            string port)
+  {
+    size_t num_workers = 5;
+    double sigma = 1; // Rewards standard deviation
+    double alpha = 0.005; // Learning Rate
+    size_t input = frameW * frameH * 3;
+    
+    while(1)
+    {
+      mat workerRewards(num_workers, 1);
+      vector<mat> epsilons(num_workers);
+      vector<Environment> environments(num_workers,
+                          Environment(host, port, environment));
+
+      #pragma omp parallel for
+      for(size_t i = 0; i < num_workers; i++)
+      {
+        mat epsilon = randn(num_actions, input);
+        mat innerModel = model + (sigma * epsilon);
+
+        epsilons[i] = epsilon;
+
+        Environment& env = environments[i];
```

Comment:
It wasn't working because Environments were overwriting each other when made in same thread :sweat_smile: So I reverted back to my earlier solution for making an environment inside worker and when worker dies, environment is closed.
About the error it was giving earlier, I found the problem was  because an environment got randomly destroyed because of death of a worker due to low memory but the keys weren't updated. So it gave error that the dictionary has expanded.
This will hopefully fix it :slightly_smiling_face: 
https://github.com/zoq/gym_tcp_api/pull/2/files#diff-ccb8997ccfef17b0369aea8581eda235R159